### PR TITLE
refactor(npm): make `NpmCache`, `CliNpmRegistryApi`, and `NpmResolution` internal to `npm::managed`

### PR DESCRIPTION
**Pull request by dsherret**
More work towards https://github.com/denoland/deno/issues/18967

### CODE REVIEW

File: cli/npm/cache_dir.rs
Comment (dsherret) [inline cli/npm/cache_dir.rs:1]: Nothing new. Just moved around.

Right now this is public to the `npm` module, but maybe in the future it won't be.

File: cli/args/mod.rs [hunk 1]
```diff
@@ -843,32 +871,15 @@ impl CliOptions {
...
-          .with_context(|| {
-            format!(
-              "failed reading lockfile '{}'",
-              lockfile.lock().filename.display()
-            )
-          })?;
-        // clear the memory cache to reduce memory usage
-        api.clear_memory_cache();
-        return Ok(Some(snapshot));
-      }
```
Comment (bartlomieju) [inline cli/args/mod.rs:868]: Has this been merged into somewhere else?

File: cli/npm/managed/mod.rs [hunk 2]
```diff
@@ -27,30 +28,213 @@ use deno_semver::npm::NpmPackageReqReference;
...
+mod cache;
 mod installer;
+mod registry;
 mod resolution;
 mod resolvers;
+mod tarball;
+
+pub enum CliNpmResolverManagedSnapshotOption {
+  ResolveFromLockfile(Arc<Mutex<Lockfile>>),
+  Provided(Option<ValidSerializedNpmResolutionSnapshot>),
```
Comment (bartlomieju) [inline cli/npm/managed/mod.rs:58]: Nitpick, and I'm not sure it's even added in this PR, but it's a bit strange that `Provided` is an option. Maybe split it into `Provided(ValidSerializedNpmResolutionSnapshot` and `None`?

File: cli/args/mod.rs (hunk 1 above)
Comment (dsherret) [reply cli/args/mod.rs:868]: Yeah, it's internal to the npm module.

File: cli/npm/managed/mod.rs (hunk 2 above)
Comment (dsherret) [reply cli/npm/managed/mod.rs:58]: It leads to some extra pattern matching. I'll rename it to `Specified`

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**